From 7ec61123f4de2c6a067622cebd689f1645683ea5 Mon Sep 17 00:00:00 2001
From: Connor Baker <ConnorBaker01@gmail.com>
Date: Sat, 26 Sep 2026 14:28:41 -0700
Subject: [PATCH] cmake: export toolkit headers and install a discoverable
 package version

Public NVSHMEM headers require CUDA headers. Publish CUDA::toolkit on both
installed targets and use the ConfigVersion filename expected by find_package.
Do not overwrite NVSHMEM_VERSION with an unset template variable.
---
 CMakeLists.txt                      | 2 +-
 cmake_config/NVSHMEMConfig.cmake.in | 1 -
 src/CMakeLists.txt                  | 4 ++--
 src/device/CMakeLists.txt           | 2 +-
 4 files changed, 4 insertions(+), 5 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index b39bcfd..05bc50a 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -165,7 +165,7 @@ configure_package_config_file(
   PATH_VARS INCLUDE_INSTALL_DIR LIB_INSTALL_DIR BIN_INSTALL_DIR)
 
 write_basic_package_version_file(
-  NVSHMEMVersion.cmake
+  NVSHMEMConfigVersion.cmake
   VERSION ${PROJECT_VERSION}
   COMPATIBILITY AnyNewerVersion)
 
diff --git a/cmake_config/NVSHMEMConfig.cmake.in b/cmake_config/NVSHMEMConfig.cmake.in
index 7a0c1f2..9d1befb 100644
--- a/cmake_config/NVSHMEMConfig.cmake.in
+++ b/cmake_config/NVSHMEMConfig.cmake.in
@@ -25,7 +25,6 @@ macro(check_required_components _NAME)
   endforeach()
 endmacro()
 
-set(NVSHMEM_VERSION @PACKAGE_VERSION@)
 
 get_filename_component(NVSHMEM_RHEL_INCL_BIN_BASE "${CMAKE_CURRENT_LIST_DIR}/../../../../../" ABSOLUTE)
 get_filename_component(NVSHMEM_DEB_INCL_BIN_BASE "${CMAKE_CURRENT_LIST_DIR}/../../../../../.." ABSOLUTE)
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 8d725e4..2eea049 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -457,7 +457,7 @@ if (NVSHMEM_BUILD_WITH_CUTLASS)
 
 endif()
 
-target_link_libraries(nvshmem_host PRIVATE CUDA::cudart_static)
+target_link_libraries(nvshmem_host PRIVATE CUDA::cudart_static PUBLIC CUDA::toolkit)
 if (CCCL_FOUND AND CUDAToolkit_VERSION_MAJOR GREATER 12)
   target_link_libraries(nvshmem_host PUBLIC CCCL::CCCL)
 endif()
@@ -935,7 +935,7 @@ install(
 )
 
 # ${CMAKE_CURRENT_BINARY_DIR}/NVSHMEMConfig.cmake
-install(FILES ${CMAKE_BINARY_DIR}/NVSHMEMVersion.cmake
+install(FILES ${CMAKE_BINARY_DIR}/NVSHMEMConfigVersion.cmake
         DESTINATION lib/cmake/nvshmem
 )
 
diff --git a/src/device/CMakeLists.txt b/src/device/CMakeLists.txt
index 1638f22..5b0fa62 100644
--- a/src/device/CMakeLists.txt
+++ b/src/device/CMakeLists.txt
@@ -183,7 +183,7 @@ else()
   configure_file(../include/non_abi/device/pt-to-pt/transfer_device.cuh.in ${CMAKE_CURRENT_SOURCE_DIR}/comm/transfer_device.cu COPYONLY)
 endif()
 
-target_link_libraries(nvshmem_device PRIVATE CUDA::cudart_static)
+target_link_libraries(nvshmem_device PRIVATE CUDA::cudart_static PUBLIC CUDA::toolkit)
 if (CCCL_FOUND AND CUDAToolkit_VERSION_MAJOR GREATER 12)
   target_link_libraries(nvshmem_device PUBLIC CCCL::CCCL)
 endif()
